test/unittests/iotools/utils/test_center_nifti.py

```diff
@@ -1,23 +1,79 @@
 from os import PathLike
 from pathlib import Path
-from typing import Iterable, List, Optional
+from typing import Iterable, Optional, Union
 from unittest.mock import patch
 
 import pytest
 
 
-def build_folder_with_specific_modalities(tmp_path: Path) -> Path:
+def build_bids_folder(tmp_path: Path) -> Path:
     bids_path = tmp_path / "BIDS"
     bids_path.mkdir()
 
-    (bids_path / "sub-1_ses-1_T1w.nii.gz").touch()
-    (bids_path / "sub-2_ses-1_T1w.nii.gz").touch()
-    (bids_path / "sub-2_ses-2_pet.nii.gz").touch()
-    (bids_path / "sub-2_ses-2_T1toto.nii.gz").touch()
+    (bids_path / "dataset_description.json").touch()
+
+    (bids_path / "sub-1" / "ses-1").mkdir(parents=True)
+    (bids_path / "sub-1" / "ses-1" / "sub-1_ses-1_T1w.nii.gz").touch()
+    (bids_path / "sub-2" / "ses-1").mkdir(parents=True)
+    (bids_path / "sub-2" / "ses-1" / "sub-2_ses-1_T1w.nii.gz").touch()
+    (bids_path / "sub-2" / "ses-2").mkdir(parents=True)
+    (bids_path / "sub-2" / "ses-2" / "sub-2_ses-2_pet.nii.gz").touch()
+    (bids_path / "sub-2" / "ses-2" / "sub-2_ses-2_T1toto.nii.gz").touch()
 
     return bids_path
 
 
+def test_handle_output_existing_files(tmp_path):
+    from clinica.iotools.utils.data_handling._centering import (
+        _handle_output_existing_files,
+    )
+    from clinica.utils.exceptions import ClinicaExistingDatasetError
+
+    output_path = tmp_path / "COPY"
+    assert _handle_output_existing_files(output_path) == output_path
+
+    output_path = build_bids_folder(tmp_path)
+    with pytest.raises(ClinicaExistingDatasetError):
+        _handle_output_existing_files(output_path)
+
+    _handle_output_existing_files(output_path, overwrite_existing_files=True)
+    assert not output_path.exists()
+
+
+def test_validate_bids_and_output_dir_equal_error(tmp_path):
+    from clinica.iotools.utils.data_handling._centering import (
+        _validate_bids_and_output_dir,
+    )
+    from clinica.utils.exceptions import ClinicaBIDSError
+
+    with pytest.raises(ClinicaBIDSError):
+        _validate_bids_and_output_dir(tmp_path / "BIDS", tmp_path / "BIDS")
+
+
+def test_validate_bids_and_output_dir_not_bids_error(tmp_path):
+    from clinica.iotools.utils.data_handling._centering import (
+        _validate_bids_and_output_dir,
+    )
+
+    (tmp_path / "BIDS").mkdir()
+    from clinica.utils.exceptions import ClinicaBIDSError
```

Comment:
I would move the import at the top of the file since it is used in lots of test functions. At least move it at the beginning of the function.